### PR TITLE
Refactor: Removed biputils dependency

### PR DESCRIPTION
**Pull request by MissingNO57**
## Proposed changes

Replaced biputils dependency with simplified implementation.

## Issues

Links to any issues resolved.

## Types of changes

What types of changes does your code introduce?
_Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change that fixes an issue)
- [x] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to stop working as expected)
- [x] Something else (e.g. tests, scripts, example, deployment, infrastructure, ...)

## Checklist

_Put an `x` in the boxes that apply._

- [x] I have read the [CONTRIBUTING](https://github.com/fetchai/cosmpy/blob/main/CONTRIBUTING.md) document.
- [x] I have based my branch, and I am making a pull request against, the `main` branch.
- [x] Checks and tests pass locally.

### If applicable

- [x] I have added tests that prove my fix is effective or that my feature works.
- [x] I have checked that code coverage does not decrease.
- [x] I have added/updated the documentations.

## Further comments

If this is a relatively large or complex change, kick off the discussion by explaining why you chose the solution you did, what alternatives you considered, etc.

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -69,6 +70,13 @@ flake8-eradicate = "==1.4.0"
 flake8-docstrings = "==1.6.0"
 flake8-isort = "==5.0"
 
+[tool.poetry.group.mnemonic]
+optional = true
+
+[tool.poetry.group.mnemonic.dependencies]
+bip-utils = "*"
```
Comment (jrriehl) [inline pyproject.toml:77]: I would vote for leaving `bip-tools` out entirely and maybe replacing it with a few tests against hard-coded mnemonics.

File: pyproject.toml [hunk 2]
```diff
@@ -54,6 +53,8 @@ check-manifest = "*"
 tox = "^3.26"
 flake8 = "==5.0.4"
 black = "^22.10"
+google-api-python-client = "*"
+googleapis-common-protos = "*"
```
Comment (ejfitzgerald) [inline pyproject.toml:57]: Do we need both? This looks like a new dependency - was it intended to replace the other Google python api?

Comment (MissingNO57) [reply pyproject.toml:56]: Looks like we need only googleapis-common-protos

Comment (MissingNO57) [reply pyproject.toml:56]: from google.api import annotations_pb2

**Review by ejfitzgerald — APPROVED** [review]

**Pull request merged**